### ISSUE TITLE
--extract-media mess the image paths up when source doc includes web image URLs

### ISSUE THREAD

**Issue by noisnemid** [jgm/pandoc#7450]
Archlinux, latest version of both OS and pandoc.

While trying to convert an HTML doc or a remote HTTP/HTTPS URL includes images with remote URLs, the `--extract-media=./media` arg will extract images to a very deep folder like this:

```
/some/folders/media/https:/pics.example.com/store/somepic1.png
```
i.e., pandoc treats the URL as local path...

**Comment by jgm** [comment]
Can you give me exact instructions for reproducing this? I've tried a few things and was not seeing this behavior.
Also, please give output of `pandoc --version`. Saying "latest version" is ambiguous. If you mean the latest version your distro has, it may not be the latest version.

**Comment by noisnemid** [comment]
> Can you give me exact instructions for reproducing this? I've tried a few things and was not seeing this behavior.
> Also, please give output of `pandoc --version`. Saying "latest version" is ambiguous. If you mean the latest version your distro has, it may not be the latest version.

```
neofetch
OS: Arch Linux x86_64 
Kernel: 5.13.4-arch1-1
```

```
$panodc --version

pandoc 2.14.0.2
Compiled with pandoc-types 1.22, texmath 0.12.3, skylighting 0.10.5.1,
citeproc 0.4.0.1, ipynb 0.1.0.1
User data directory: /home/br/.local/share/pandoc
Copyright (C) 2006-2021 John MacFarlane. Web:  https://pandoc.org
This is free software; see the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.
```

TEST CASE:
```
~/test:ls
test.html
~/test:cat test.html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>test</title>
</head>
<body>
    <h1>test</h1>
    <p>para para para</p>
    <img src="https://pic3.zhimg.com/80/81787fffb91015015bbe03f8ad566071_720w.jpg?source=1940ef5c">
</body>
</html>⏎                                                                                                                                                                    ~/test:pandoc test.html -o test.rst --extract-media=./media
~/test:ls -R
.:
media  test.html  test.rst

./media:
https:

'./media/https:':
pic3.zhimg.com

'./media/https:/pic3.zhimg.com':
80

'./media/https:/pic3.zhimg.com/80':
'81787fffb91015015bbe03f8ad566071_720w.jpg?source=1940ef5c'
~/test:

```
[test.zip](https://github.com/jgm/pandoc/files/6859661/test.zip)

**Comment by noisnemid** [comment]
> Can you give me exact instructions for reproducing this? I've tried a few things and was not seeing this behavior.
> Also, please give output of `pandoc --version`. Saying "latest version" is ambiguous. If you mean the latest version your distro has, it may not be the latest version.

```
neofetch
OS: Arch Linux x86_64 
Kernel: 5.13.4-arch1-1
```

```
$panodc --version

pandoc 2.14.0.2
Compiled with pandoc-types 1.22, texmath 0.12.3, skylighting 0.10.5.1,
citeproc 0.4.0.1, ipynb 0.1.0.1
User data directory: /home/****/.local/share/pandoc
Copyright (C) 2006-2021 John MacFarlane. Web:  https://pandoc.org
This is free software; see the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.
```

TEST CASE:
```
~/test:ls
test.html
~/test:cat test.html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>test</title>
</head>
<body>
    <h1>test</h1>
    <p>para para para</p>
    <img src="https://pic3.zhimg.com/80/81787fffb91015015bbe03f8ad566071_720w.jpg?source=1940ef5c">
</body>
</html>⏎                                                                                                                                                                    ~/test:pandoc test.html -o test.rst --extract-media=./media
~/test:ls -R
.:
media  test.html  test.rst

./media:
https:

'./media/https:':
pic3.zhimg.com

'./media/https:/pic3.zhimg.com':
80

'./media/https:/pic3.zhimg.com/80':
'81787fffb91015015bbe03f8ad566071_720w.jpg?source=1940ef5c'
~/test:

```
[test.zip](https://github.com/jgm/pandoc/files/6859661/test.zip)

**Comment by noisnemid** [comment]
I've confirmed that:
This is a old bug, which has been already fixed in the 2.14.1 release.
This is a problem of Archlinux's package community verison, not the source's version.
Next time I will install the github's version instead of the pacman's version for daily using and testing.
My fault.

Conculsion:
After I download the release package of 2.14.1 from github's release tag, this issue doesn't exist.

But, there's another old small problem:

the picture was not named in sequenced numbers.

Don't know whether it's an issue.

**Comment by jgm** [comment]
We switched to using sha1 hashes to name downloaded images.
